src/common/config/store/configstore.go

```diff
@@ -0,0 +1,116 @@
+package store
+
+import (
+	"errors"
+	"fmt"
+	"github.com/goharbor/harbor/src/common/config/metadata"
+	"github.com/goharbor/harbor/src/common/config/store/driver"
+	"github.com/goharbor/harbor/src/common/http/modifier/auth"
+	"github.com/goharbor/harbor/src/common/utils/log"
+	"sync"
+)
+
+// Driver ...
+type Driver interface {
+	// Load - load config item from config driver
+	Load() (map[string]interface{}, error)
+	// Save - save config item into config driver
+	Save(cfg map[string]interface{}) error
+}
+
+// ConfigStore - the config data store
+type ConfigStore struct {
+	cfgDriver Driver
+	cfgValues sync.Map
+}
+
+// NewDBConfigStore - Create config store based on database and env
+func NewDBConfigStore() *ConfigStore {
+	return &ConfigStore{cfgDriver: &driver.Database{}}
+}
+
+// NewHTTPConfigStore - Create config store based on Rest API
+func NewHTTPConfigStore(url, secret string) *ConfigStore {
+	secAuth := auth.NewSecretAuthorizer(secret)
+	return &ConfigStore{cfgDriver: driver.NewRESTDriver(url, secAuth)}
+}
+
+// Get - Get config data from current store
+func (c *ConfigStore) Get(key string) (*metadata.ConfigureValue, error) {
+	if value, ok := c.cfgValues.Load(key); ok {
```

Comment:
this load are so confusing with the Load() of ConfigStore().

Reply:
the cfgValues are sync.Map, the Load is method, same function as Get, it is the API